File: app/src/main/java/com/bureng/robocoinx/contract/LoginContract.java
package com.bureng.robocoinx.contract;

import android.content.Context;

import com.bureng.robocoinx.model.request.SignUpRequest;
import com.bureng.robocoinx.model.view.ProfileView;

public interface LoginContract {
    interface View{
        void showMessage(String message);
        void validateInput();
        void goHome(ProfileView profileView);
        void setCaptchaNet(String captchaNet);
    }
    interface Presenter{
        void login(Context ctx, String email, String password);
        void signUp(Context ctx, SignUpRequest request);
        void getCaptchaNet(String fingerprint);
    }
}

File: app/src/main/java/com/bureng/robocoinx/contract/LoginContract.kt
package com.bureng.robocoinx.contract

import android.app.Activity
import android.content.Context
import com.bureng.robocoinx.model.view.ProfileView

interface LoginContract {
    interface View {
        fun showMessage(message: String, type: Int)
        fun flagHome()
        fun goHome(profileView: ProfileView)
        fun showProgressBar(rawLoading: Int?)
        fun hideProgressBar()
    }

    interface Presenter {
        fun login(act: Activity, ctx: Context, email: String, password: String)
        fun getHomeResponse(ctx: Context)
    }
}